analyses/molecular-subtyping-pathology/utils/recode_integrated_diagnosis.R

```diff
@@ -0,0 +1,70 @@
+#' sample_id,tumor_descriptor,pathology_free_text_diagnosis,broad_histology,short_histology,
+#' molecular_subtype,integrated_diagnosis
+#' @param pathology_diagnois : A term to filter histologies_df to samples that will be recoded
+#' @param include_path_free_text_dx_terms : Regex term to match with values 
+#' in pathology_free_text_diagnosis in histologies_df to be included in the recoding analysis
+#' @param exclude_path_free_text_dx_terms : Default = NULL; Regex term to match with values in 
+#' pathology_free_text_diagnosis in histologies_df to be excluded from the recoding analysis
+#' @param old_integrated_diagnosis_term : Old term in integrated_diagnosis that should be replaced 
+#' by replace_integrated_diagnosis_term
+#' @param replace_integrated_diagnosis_term : New term for integrated_diagnosis that will replace 
+#' an old integrated_term
+
+
+recode_integrated_diagnosis <- function(histologies_df,
+                                        pathology_diagnosis,
+                                        include_path_free_text_dx_terms,
+                                        exclude_path_free_text_dx_terms=NULL,
+                                        old_integrated_diagnosis_term,
+                                        replace_integrated_diagnosis_term){
+  subtype_include_df <- histologies_df %>%
+    # We are only concerned with samples where the pathology_free_text_diagnosis
+    # contains the include term
+    dplyr::filter(pathology_diagnosis == pathology_diagnosis,
+                  grepl(include_path_free_text_dx_terms,pathology_free_text_diagnosis)) %>%
+    # Retain only relevant identifier and disease label columns
+    dplyr::select(Kids_First_Biospecimen_ID,
+                  Kids_First_Participant_ID,
+                  sample_id,
+                  tumor_descriptor,
+                  pathology_free_text_diagnosis,
+                  broad_histology,
+                  short_histology,
+                  molecular_subtype,
+                  integrated_diagnosis) %>%
+    # To smooth the way for string detection for the pathology free text, we
+    # add a column where all of the text is converted to lowercase
+    dplyr::mutate(pathology_free_text_dx_lower = 
+                    stringr::str_to_lower(pathology_free_text_diagnosis)) %>%
+    # String detection in pathology free text per the table
+    dplyr::mutate(
+      integrated_diagnosis = dplyr::case_when(
+        # if NA add the new integrated diagnosis
+        is.na(integrated_diagnosis) ~ replace_integrated_diagnosis_term,
+        # if there is an existing integrated_diagnosis then replace with new integrated_diagnosis
+        stringr::str_detect(pathology_free_text_dx_lower,
+                            include_path_free_text_dx_terms) ~ str_replace(integrated_diagnosis,old_integrated_diagnosis_term,replace_integrated_diagnosis_term),
+      ),
+      Notes = "Updated via OpenPBTA subtyping from pathology_free_text_diagnosis"
```

Comment:
```suggestion
      Notes = "Updated via OpenPBTA subtyping and pathology_free_text_diagnosis"
```

I would suggest adding this as an _and_ statement, since the initial update was via OpenPBTA subtyping. For these samples, I am not clear how we are going to add them to a new file and remove the old values. In my head, I was thinking this would update the values in the old file so we wouldn't have to do an rbind of files later plus a removal of values. Thoughts @kgaonkar6 and @jaclyn-taroni ?